File: backend/src/main/kotlin/com/cyborgstudios/gamebuilder/orchestration/domain/repository/ProjectRepository.kt
package com.cyborgstudios.gamebuilder.orchestration.domain.repository

import com.cyborgstudios.gamebuilder.orchestration.domain.model.Project
import com.cyborgstudios.gamebuilder.orchestration.domain.model.ProjectStatus
import java.util.UUID

interface ProjectRepository {

    fun save(project: Project): Project

    fun findById(id: UUID): Project?

    fun updateStatus(id: UUID, status: ProjectStatus): Boolean
}

File: backend/src/gamebuilder/orchestration/domain/repository/project_repository.py
from typing import Protocol
from uuid import UUID

from gamebuilder.orchestration.domain.model.project import Project
from gamebuilder.orchestration.domain.model.project_status import ProjectStatus


class ProjectRepository(Protocol):
    async def save(self, project: Project) -> Project: ...

    async def find_by_id(self, project_id: UUID) -> Project | None: ...

    async def list_recent(self) -> list[Project]: ...

    async def update_status(self, project_id: UUID, status: ProjectStatus) -> None: ...

    async def delete(self, project_id: UUID) -> None: ...
